### PR TITLE
add PCI bus scanning capabilities

### PR DESCRIPTION
**Pull request by Rwantare**
Even though I was unable to get PCI configured, some of the functionality in that effort was useful and can be merged.

Signed-off-by: Titus Rwantare <titusr@google.com>

### CODE REVIEW

**Review by daym — DISMISSED** [review]
Looks good, thanks!

File: src/arch/x86/x86_64/src/lib.rs [hunk 1]
```diff
@@ -2,13 +2,14 @@
 #![feature(lang_items, start)]
 #![no_std]
 #![feature(global_asm)]
-#![deny(warnings)]
+// #![deny(warnings)]
```
Comment (rjoleary) [inline src/arch/x86/x86_64/src/lib.rs:5]: Remove this line.

File: src/arch/x86/x86_64/Cargo.toml [hunk 2]
```diff
@@ -18,3 +18,4 @@ opt-level = 'z'  # Optimize for size.
 ioport = []
 bzimage = []
 acpi = []
+pci = []
```
Comment (rjoleary) [inline src/arch/x86/x86_64/Cargo.toml:21]: Add newline at end of file.

File: src/arch/x86/x86_64/src/pci.rs [hunk 3]
```diff
@@ -0,0 +1,287 @@
+use core::ptr;
+use model::*;
```
Comment (rjoleary) [inline src/arch/x86/x86_64/src/pci.rs:2]: Add newline after this line.

File: src/arch/x86/x86_64/src/pci.rs [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+fn smn_write(a: u32, v: u32) {
+    unsafe {
+        outl(0xcf8, 0x800000b8);
+        outl(0xcfc, a);
+        outl(0xcf8, 0x800000bc);
+        outl(0xcfc, v);
+    }
+}
+/// Write 32 bits to port
```
Comment (rjoleary) [inline src/arch/x86/x86_64/src/pci.rs:71]: Add newline before this line.

**Pull request closed without merging**